src/components/Datatable/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,161 @@
+/* eslint-disable react/boolean-prop-naming */
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { unfold } from 'ramda';
+
+import { getColor, pxToRem } from '../../../utils/helpers';
+import { FlexContainer } from '../../FlexContainer';
+import { SSCIconNames } from '../../../theme/icons/icons.enums';
+import { Spinner } from '../../Spinner';
+import { Label } from '../../forms/Label';
+import { PaginationProps } from './Pagination.types';
+import PageButton, { ActivePage } from './PageButton';
+import NavButton from './NavButton';
+import { StyledNumber } from '../../Filters/components/Number/Number';
+
+const PaginationWrapper = styled(FlexContainer)`
+  background-color: ${getColor('graphite3H')};
+  padding: ${pxToRem(16)};
+`;
+
+const SpinnerContainer = styled(FlexContainer)`
+  min-width: ${pxToRem(48)};
+`;
+
+const Ellipsis = styled.div`
+  display: inline-block;
+  width: ${pxToRem(48)};
+  height: ${pxToRem(32)};
+  padding: ${pxToRem(8, 16)};
+  margin: ${pxToRem(0, 4)};
+  color: ${getColor('graphite2B')};
+  background-color: ${getColor('transparent')};
+`;
+
+const PageInputLabel = styled(Label)`
+  margin: ${pxToRem(6, 16)};
+`;
+const PageInput = styled(StyledNumber)`
+  width: ${pxToRem(72)};
+  height: ${pxToRem(32)};
+  padding: ${pxToRem(8, 16)};
+`;
+
+const Pagination: React.FC<PaginationProps> = ({
+  onGoToPage,
+  onPreviousPage,
+  onNextPage,
+  pageCount,
+  canPreviousPage,
+  canNextPage,
+  pageIndex,
+  isLoading,
+  numPageButtons = 8,
+}) => {
+  const generatePages = (start, end) =>
+    unfold((p) => (p > end ? false : [p, p + 1]), start);
+
+  const currentPage = pageIndex + 1;
+
+  const showLeftEllipsis = currentPage > numPageButtons - 3;
+  const showRightEllipsis = currentPage < pageCount - (numPageButtons - 4);
+
+  const threshold = numPageButtons - 3;
+
+  const startPage =
+    currentPage <= threshold
+      ? 2
+      : currentPage > pageCount - threshold
+      ? pageCount - threshold
+      : currentPage - Math.ceil((numPageButtons - 4 - 1) / 2);
+  const endPage =
+    currentPage <= threshold
+      ? numPageButtons - 2
+      : currentPage > pageCount - threshold
+      ? pageCount - 1
+      : currentPage + Math.floor((numPageButtons - 4 - 1) / 2);
+
+  const middlePageButtons = generatePages(startPage, endPage);
+
+  const renderPageButton = (page) => {
```

Comment:
this method can be taken out of the component

Reply:
at the cost of adding two variables to it but no problem ...